src/main/java/uk/gov/pay/ledger/transaction/search/common/TransactionSearchParams.java

```diff
@@ -304,6 +308,33 @@ private Long getOffset() {
         return offset;
     }
 
+    private List<String> getTransactionStatesForStatuses(CommaDelimitedSetParameter paymentStates) {
+        return paymentStates.getParameters().stream()
+                .map(TransactionState::getStatesForStatus)
+                .flatMap(List::stream)
+                .map(TransactionState::name)
+                .collect(Collectors.toList());
+    }
```

Comment:
Given this fancy stream logic, it would be good to assert on generated links as well. (or some sort of unit test for it)

Reply:
the generated links need to just echo the input values, whereas this is mapping from input value to the TransactionState. So I'm not sure what could go wrong with links? Unless I'm misunderstanding you

Reply:
No, I misread. I thought it was being used in generating links as well :doh: